apl-core/src/main/java/com/apollocurrency/aplwallet/apl/alias/service/AliasServiceImpl.java

```diff
@@ -0,0 +1,184 @@
+    ) {
+        this.aliasTable = aliasTable;
+        this.offerTable = offerTable;
+        this.blockchain = blockchain;
+        this.converter = converter;
+    }
+
+    @Inject
+    public AliasServiceImpl(
+        final AliasTable aliasTable,
+        final AliasOfferTable offerTable,
+        final Blockchain blockchain
+    ) {
+        this.aliasTable = aliasTable;
+        this.offerTable = offerTable;
+        this.blockchain = blockchain;
+        this.converter = new IteratorToStreamConverter<>();
+    }
+
+    @Override
+    public int getCount() {
+        return aliasTable.getCount();
+    }
+
+    @Override
+    public int getAccountAliasCount(long accountId) {
+        return aliasTable.getCount(new DbClause.LongClause("account_id", accountId));
+    }
+
+    @Override
+    public Stream<Alias> getAliasesByOwner(long accountId, int from, int to) {
+        final DbIterator<Alias> aliasIterator = aliasTable.getManyBy(
+            new DbClause.LongClause("account_id", accountId), from, to
+        );
+
+        return converter.convert(aliasIterator);
+    }
+
+    @Override
+    public Alias getAliasByName(String aliasName) {
+        return aliasTable.getBy(new DbClause.StringClause("alias_name_lower", aliasName.toLowerCase()));
+    }
+
+    @Override
+    public Stream<Alias> getAliasesByNamePattern(String aliasName, int from, int to) {
+        final DbIterator<Alias> aliasIterator = aliasTable.getManyBy(
+            new DbClause.LikeClause("alias_name_lower", aliasName.toLowerCase()), from, to
+        );
+
+        return converter.convert(aliasIterator);
+    }
+
+    @Override
+    public Alias getAliasById(long id) {
+        return aliasTable.getAlias(id);
+    }
+
+    @Override
+    public void deleteAlias(final String aliasName) {
+        final Alias alias = getAliasByName(aliasName);
+        final AliasOffer offer = offerTable.getOffer(alias);
+        final int height = blockchain.getHeight();
+        if (offer != null) {
+            offer.setPriceATM(Long.MAX_VALUE);
+            offer.setHeight(height);
+            offerTable.deleteAtHeight(offer, height);
+        }
+        alias.setHeight(height);
+        aliasTable.deleteAtHeight(alias, height);
+    }
+
+    @Override
+    public void addOrUpdateAlias(Transaction transaction, MessagingAliasAssignment attachment) {
+        Alias alias = getAliasByName(attachment.getAliasName());
+        if (alias == null) {
+            alias = new Alias(transaction, attachment, blockchain.getHeight(), blockchain.getLastBlockTimestamp());
+        } else {
+            alias.setHeight(blockchain.getHeight());
+            alias.setAccountId(transaction.getSenderId());
+            alias.setAliasURI(attachment.getAliasURI());
+            alias.setTimestamp(blockchain.getLastBlockTimestamp());
+        }
+        aliasTable.insert(alias);
+    }
+
+    @Override
+    public void sellAlias(Transaction transaction, MessagingAliasSell attachment) {
```

Comment:
It would be good to add check input to prevent NPE - Objects.requireNonNull(..)

Reply:
done